curvefs/src/client/fuse_client.cpp

```diff
@@ -895,38 +899,63 @@ CURVEFS_ERROR FuseClient::FuseOpRename(fuse_req_t req, fuse_ino_t parent,
         return CURVEFS_ERROR::NAME_TOO_LONG;
     }
 
-    auto renameOp =
-        RenameOperator(fsInfo_->fsid(), fsInfo_->fsname(),
-                       parent, name, newparent, newname,
-                       dentryManager_, inodeManager_, metaClient_, mdsClient_,
-                       option_.enableMultiMountPointRename);
+    auto renameOp = RenameOperator(fsInfo_->fsid(), fsInfo_->fsname(), parent,
+        name, newparent, newname, dentryManager_, inodeManager_, metaClient_,
+        mdsClient_, option_.enableMultiMountPointRename);
 
-    curve::common::LockGuard lg(renameMutex_);
     CURVEFS_ERROR rc = CURVEFS_ERROR::OK;
-    VLOG(3) << "FuseOpRename [start]: " << renameOp.DebugString();
-    RETURN_IF_UNSUCCESS(GetTxId);
-    RETURN_IF_UNSUCCESS(Precheck);
-    RETURN_IF_UNSUCCESS(RecordOldInodeInfo);
-    // Do not move LinkDestParentInode behind CommitTx.
-    // If so, the nlink will be lost when the machine goes down
-    RETURN_IF_UNSUCCESS(LinkDestParentInode);
-    RETURN_IF_UNSUCCESS(PrepareTx);
-    RETURN_IF_UNSUCCESS(CommitTx);
-    VLOG(3) << "FuseOpRename [success]: " << renameOp.DebugString();
-    // Do not check UnlinkSrcParentInode, beause rename is already success
-    renameOp.UnlinkSrcParentInode();
-    renameOp.UnlinkOldInode();
-    if (parent != newparent) {
-        renameOp.UpdateInodeParent();
-    }
-    renameOp.UpdateInodeCtime();
-    renameOp.UpdateCache();
+    if (common::FLAGS_TxVersion == 1) {
+        curve::common::LockGuard lg(renameMutex_);
+        VLOG(3) << "FuseOpRename [start]: " << renameOp.DebugString();
+        RETURN_IF_UNSUCCESS(GetTxId);
+        RETURN_IF_UNSUCCESS(Precheck);
+        RETURN_IF_UNSUCCESS(RecordOldInodeInfo);
+        // Do not move LinkDestParentInode behind CommitTx.
+        // If so, the nlink will be lost when the machine goes down
+        RETURN_IF_UNSUCCESS(LinkDestParentInode);
```

Comment:
If the subsequent operation fails, is it necessary to perform the unlink operation?

Reply:
Maybe, nilink is an extremely complex processing logic on the client side, and v2 maintains the same processing flow as v1.

Reply:
The link op on src and dst parentInode have no affect if failed.